drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+static int flash_stm32_ospi_write(const struct device *dev, off_t addr,
+				  const void *data, size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	struct flash_stm32_ospi_data *dev_data = DEV_DATA(dev);
+	int ret = 0;
+
+	if (!ospi_address_is_valid(dev, addr, size)) {
+		LOG_DBG("Error: address or size exceeds expected values: "
+			"addr 0x%lx, size %zu", (long)addr, size);
+		return -EINVAL;
+	}
+
+	/* write non-zero size */
+	if (size == 0) {
+		return 0;
+	}
+
+	/* page program for STR or DTR mode */
+	OSPI_RegularCmdTypeDef cmd_pp = ospi_prepare_cmd(dev_cfg->data_mode, dev_cfg->data_rate);
+	cmd_pp.DummyCycles = 0;
+
+	LOG_INF("OSPI: write %u data", size);
+	ospi_lock_thread(dev);
+
+	while ((size > 0) && (ret == 0)) {
+		size_t to_write = size;
+
+		ret = stm32_ospi_mem_ready(&dev_data->hospi,
+						 dev_cfg->data_mode, dev_cfg->data_rate);
+		if (ret != 0) {
+			break;
+		}
+
+		ret = stm32_ospi_write_enable(&dev_data->hospi,
+						    dev_cfg->data_mode, dev_cfg->data_rate);
+		if (ret != 0) {
+			break;
+		}
+		/* Don't write more than a page. */
+		if (to_write >= SPI_NOR_PAGE_SIZE) {
+			to_write = SPI_NOR_PAGE_SIZE;
+		}
+
+		/* Don't write across a page boundary */
+		if (((addr + to_write - 1U) / SPI_NOR_PAGE_SIZE)
+		    != (addr / SPI_NOR_PAGE_SIZE)) {
+			to_write = SPI_NOR_PAGE_SIZE -
+						(addr % SPI_NOR_PAGE_SIZE);
+		}
+
+		cmd_pp.Instruction = (dev_cfg->data_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_4B_PP
+				: SPI_NOR_OCMD_PAGE_PRG;
+		cmd_pp.Address = addr;
+		ret = ospi_write_access(dev, &cmd_pp, data, to_write);
+		if (ret != 0) {
+			break;
+		}
+
+		size -= to_write;
+		data = (const uint8_t *)data + to_write;
+		addr += to_write;
+
+		/* Configure automatic polling mode to wait for end of program */
+		ret = stm32_ospi_mem_ready(&dev_data->hospi,
+						 dev_cfg->data_mode, dev_cfg->data_rate);
+		if (ret != 0) {
+			break;
+		}
+	}
+
+	ospi_unlock_thread(dev);
+
+	return ret;
+}
+
+static const struct flash_parameters flash_stm32_ospi_parameters = {
+	.write_block_size = 1,
+	.erase_value = 0xff
+};
+
+static const struct flash_parameters *
+flash_stm32_ospi_get_parameters(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return &flash_stm32_ospi_parameters;
+}
+
+static void flash_stm32_ospi_isr(const struct device *dev)
+{
+	struct flash_stm32_ospi_data *dev_data = DEV_DATA(dev);
+
+	HAL_OSPI_IRQHandler(&dev_data->hospi);
+}
+
+/* weak function required for HAL compilation */
+__weak HAL_StatusTypeDef HAL_DMA_Abort_IT(DMA_HandleTypeDef *hdma)
+{
+	return HAL_OK;
+}
+
+/* weak function required for HAL compilation */
+__weak HAL_StatusTypeDef HAL_DMA_Abort(DMA_HandleTypeDef *hdma)
+{
+	return HAL_OK;
+}
+
+/*
+ * Transfer Error callback.
+ */
+void HAL_OSPI_ErrorCallback(OSPI_HandleTypeDef *hospi)
+{
+	struct flash_stm32_ospi_data *dev_data =
+		CONTAINER_OF(hospi, struct flash_stm32_ospi_data, hospi);
+
+	LOG_DBG("Error cb");
```

Comment:
> LOG_DBG("Error cb");

Can we clarify the error a bit more? If not, then maybe it is not worth having the log output here at all?